### PR TITLE
Rename `instrumentation` to `stopwatch`

### PR DESCRIPTION
**Pull request by matanlurey**
... since that's what the code is/does.

Partial work towards https://github.com/flutter/flutter/issues/126009.

### CODE REVIEW

File: flow/stopwatch_unittests.cc [hunk 1]
```diff
@@ -2,8 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#include "flutter/flow/instrumentation.h"
-#include "gmock/gmock.h"
+#include "flutter/flow/stopwatch.h"
+#include "gmock/gmock.h"  // IWYU pragma: keep
```
Comment (jonahwilliams) [inline flow/stopwatch_unittests.cc:6]: @dnfield was the issue that IWYU didnt understand all of the gn options and so tests didn't work well with it?

Anyway, unsure if we want to start adding IWYU pragmas until we understand how many we'd need

**Review by jonahwilliams — APPROVED** [review]
LGTM w/ question

File: flow/stopwatch_unittests.cc (hunk 1 above)
Comment (matanlurey) [reply flow/stopwatch_unittests.cc:6]: Ack. This was more of a "hmm, what's going on here".

From my understanding, gmock specifically needs a pragma (I see the Chromium codebase doing the same thing).

I'm going to merge, but if we end up wanting to remove this I have no qualms.

Comment (jonahwilliams) [reply flow/stopwatch_unittests.cc:6]: its its only for gmock and only for a few places I think thats fine. I just dont want IWYU pragmas on half of our includes 😄

Comment (matanlurey) [reply flow/stopwatch_unittests.cc:6]: Ack. I'll run that experiment later this week to see how bad it would be.

**Comment by matanlurey** [comment]
Merging ignoring `flutter-gold` as this doesn't impact goldens anyway.

File: flow/stopwatch_unittests.cc (hunk 1 above)
Comment (dnfield) [reply flow/stopwatch_unittests.cc:6]: I do not think we're using IWYU , are we?

Comment (jonahwilliams) [reply flow/stopwatch_unittests.cc:6]: I was asking about the previous issue with trying to use IWYU that you mentioned.

**Pull request merged**